src/base/static/state/ducks/ui.js

```diff
@@ -1,130 +1,109 @@
-const UPDATE_MAP_CENTERPOINT_VISIBILITY =
-  "ui/UPDATE_MAP_CENTERPOINT_VISIBILITY";
-const UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY =
-  "ui/UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY";
+const UPDATE_CURRENT_TEMPLATE = "ui/UPDATE_CURRENT_TEMPLATE";
 const UPDATE_EDIT_MODE_TOGGLED = "ui/UPDATE_EDIT_MODE_TOGGLED";
+const UPDATE_UI_VISIBILITY = "ui/UPDATE_UI_VISIBILITY";
+const UPDATE_ACTIVE_PAGE = "ui/UPDATE_ACTIVE_PAGE";
+const UPDATE_CONTENT_PANEL_COMPONENT = "ui/UPDATE_CONTENT_PANEL_COMPONENT";
+const UPDATE_LAYOUT = "ui/UPDATE_LAYOUT";
 
 // Action creators:
-export function setContentPanel(isOpen) {
-  return { type: SET_UI_CONTENT_PANEL, payload: isOpen };
-}
-export function setRightSidebar(isExpanded) {
-  return { type: SET_UI_RIGHT_SIDEBAR, payload: isExpanded };
+export function updateCurrentTemplate(templateName) {
+  return { type: UPDATE_CURRENT_TEMPLATE, payload: templateName };
 }
-export function setLeftSidebar(status) {
-  return { type: SET_UI_LEFT_SIDEBAR, payload: status };
+export function updateEditModeToggled(isToggled) {
+  return { type: UPDATE_EDIT_MODE_TOGGLED, payload: isToggled };
 }
-export function setCurrentTemplate(templateName) {
-  return { type: SET_CURRENT_TEMPLATE, payload: templateName };
+export function updateUIVisibility(uiComponentName, isVisible) {
+  return {
+    type: UPDATE_UI_VISIBILITY,
+    payload: { uiComponentName, isVisible },
+  };
 }
-export function setAddPlaceButtonVisibility(isVisible) {
-  return { type: UPDATE_ADD_PLACE_BUTTON_VISIBILITY, payload: isVisible };
+export function updateActivePage(pageSlug) {
+  return {
+    type: UPDATE_ACTIVE_PAGE,
+    payload: pageSlug,
+  };
 }
-export function setMapCenterpointVisibility(isVisible) {
-  return { type: UPDATE_MAP_CENTERPOINT_VISIBILITY, payload: isVisible };
+export function updateContentPanelComponent(componentName) {
+  return {
+    type: UPDATE_CONTENT_PANEL_COMPONENT,
+    payload: componentName,
+  };
 }
-export function setGeocodeAddressBarVisibility(isVisible) {
-  return { type: UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY, payload: isVisible };
-}
-export function updateEditModeToggled(isToggled) {
-  return { type: UPDATE_EDIT_MODE_TOGGLED, payload: isToggled };
+export function updateLayout() {
+  return {
+    type: UPDATE_LAYOUT,
+    payload: getLayout(),
+  };
 }
 
 // Reducers:
 const INITIAL_STATE = {
-  isContentPanelOpen: false,
-  isRightSidebarExpanded: false,
-  isLeftSidebarExpanded: false,
-  leftSidebarComponent: undefined,
+  activePageSlug: null,
+  uiVisibility: {
+    addPlaceButton: false,
+    contentPanel: false,
+    mapCenterpoint: false,
+    spotlightMask: false,
+    rightSidebar: false,
+  },
+  contentPanelComponent: null,
+  leftSidebarComponent: null,
   currentTemplate: "map",
-  isAddPlaceButtonVisible: false,
-  isMapCenterpointVisible: false,
-  isGeocodeAddressBarVisible: false,
   isEditModeToggled: false,
+  // Currently either "desktop" or "mobile"
+  layout: getLayout(),
```

Comment:
That's a good point-- I wasn't sure whether this should be a piece of global state. Note thought that we do have a `resize` listener on the `window` object which updates this state, so I think it will not go stale.

I figured that since so many components depend on knowing the overall layout, it would be worth keeping this value in redux.

Reply:
Sounds good :+1: